### PR TITLE
Manually update com.google.jimfs:jimfs from 1.2 to 1.3.0

### PR DESCRIPTION
**Pull request by scrawfor99**
### Description
Manually update com.google.jimfs:jimfs from 1.2 to 1.3.0 since #8579 failed.

### Check List
- [ ] ~New functionality includes testing.~
  - [ ] ~All tests pass~
- [ ] ~New functionality has been documented.~
  - [ ] ~New functionality has javadoc added~
- [x] Commits are signed per the DCO using --signoff
- [x] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: distribution/tools/keystore-cli/build.gradle [hunk 1]
```diff
@@ -34,7 +34,7 @@ dependencies {
   compileOnly project(":server")
   compileOnly project(":libs:opensearch-cli")
   testImplementation project(":test:framework")
-  testImplementation 'com.google.jimfs:jimfs:1.2'
+  testImplementation 'com.google.jimfs:jimfs:1.3'
```
Comment (Rishikesh1159) [inline distribution/tools/keystore-cli/build.gradle:37]: can you please replace `testImplementation 'com.google.jimfs:jimfs:1.3`' with 
`testImplementation 'com.google.jimfs:jimfs:1.3.0'` here.

It is throwing error that it is not able to pull anything from `https://repo.maven.apache.org/maven2/com/google/jimfs/jimfs/1.3/jimfs-1.3.pom` instead we should pull from `https://repo.maven.apache.org/maven2/com/google/jimfs/jimfs/1.3.0/jimfs-1.3.0.pom`

**Review by Rishikesh1159 — CHANGES_REQUESTED** [review]

File: distribution/tools/keystore-cli/build.gradle (hunk 1 above)
Comment (Rishikesh1159) [reply distribution/tools/keystore-cli/build.gradle:37]: Similarly replace bumping 1.3 with bumping 1.3.0 in other file changes as well.

Comment (scrawfor99) [reply distribution/tools/keystore-cli/build.gradle:37]: Will do

**Comment by owaiskazi19** [comment]
Duplicate of https://github.com/opensearch-project/OpenSearch/pull/8567 & https://github.com/opensearch-project/OpenSearch/pull/8579. We can handle the test failures in the same dependabot PRs.

**Comment by scrawfor99** [comment]
Flaky tests:

[org.opensearch.index.shard.RemoteStoreRefreshListenerTests.testRefreshSuccessOnSecondAttempt](https://build.ci.opensearch.org/job/gradle-check/19881/testReport/junit/org.opensearch.index.shard/RemoteStoreRefreshListenerTests/testRefreshSuccessOnSecondAttempt/)
    [org.opensearch.index.shard.RemoteStoreRefreshListenerTests.testRefreshSuccessOnThirdAttempt](https://build.ci.opensearch.org/job/gradle-check/19881/testReport/junit/org.opensearch.index.shard/RemoteStoreRefreshListenerTests/testRefreshSuccessOnThirdAttempt/)
    [org.opensearch.index.shard.RemoteStoreRefreshListenerTests.testRefreshSuccessOnSecondAttempt](https://build.ci.opensearch.org/job/gradle-check/19881/testReport/junit/org.opensearch.index.shard/RemoteStoreRefreshListenerTests/testRefreshSuccessOnSecondAttempt_2/)
    [org.opensearch.index.shard.RemoteStoreRefreshListenerTests.testRefreshSuccessOnThirdAttempt](https://build.ci.opensearch.org/job/gradle-check/19881/testReport/junit/org.opensearch.index.shard/RemoteStoreRefreshListenerTests/testRefreshSuccessOnThirdAttempt_2/)
    [org.opensearch.index.shard.RemoteStoreRefreshListenerTests.testRefreshSuccessOnSecondAttempt](https://build.ci.opensearch.org/job/gradle-check/19881/testReport/junit/org.opensearch.index.shard/RemoteStoreRefreshListenerTests/testRefreshSuccessOnSecondAttempt_3/)
    [org.opensearch.index.shard.RemoteStoreRefreshListenerTests.testRefreshSuccessOnThirdAttempt](https://build.ci.opensearch.org/job/gradle-check/19881/testReport/junit/org.opensearch.index.shard/RemoteStoreRefreshListenerTests/testRefreshSuccessOnThirdAttempt_3/)
    [org.opensearch.index.shard.RemoteStoreRefreshListenerTests.testRefreshSuccessOnSecondAttempt](https://build.ci.opensearch.org/job/gradle-check/19881/testReport/junit/org.opensearch.index.shard/RemoteStoreRefreshListenerTests/testRefreshSuccessOnSecondAttempt_4/)
    [org.opensearch.index.shard.RemoteStoreRefreshListenerTests.testRefreshSuccessOnThirdAttempt](https://build.ci.opensearch.org/job/gradle-check/19881/testReport/junit/org.opensearch.index.shard/RemoteStoreRefreshListenerTests/testRefreshSuccessOnThirdAttempt_4/)

```
java.lang.AssertionError: expected:<0> but was:<1>
	at __randomizedtesting.SeedInfo.seed([EC329868F74DF950:F8EB60D17522E01C]:0)
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:647)
	at org.junit.Assert.assertEquals(Assert.java:633)
	at org.opensearch.index.shard.RemoteStoreRefreshListenerTests.lambda$testRefreshSuccessOnSecondAttempt$2(RemoteStoreRefreshListenerTests.java:269)
	at org.opensearch.test.OpenSearchTestCase.assertBusy(OpenSearchTestCase.java:1085)
	at org.opensearch.test.OpenSearchTestCase.assertBusy(OpenSearchTestCase.java:1058)
	at org.opensearch.index.shard.RemoteStoreRefreshListenerTests.testRefreshSuccessOnSecondAttempt(RemoteStoreRefreshListenerTests.java:269)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1750)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:938)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:974)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:988)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
	at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
	at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
	at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
	at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:817)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:468)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:947)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:832)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:883)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:894)
	at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
	at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
	at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
	at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
	at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
	at java.base/java.lang.Thread.run(Thread.java:1623)
	Suppressed: java.lang.AssertionError: expected:<0> but was:<1>
		at org.junit.Assert.fail(Assert.java:89)
		at org.junit.Assert.failNotEquals(Assert.java:835)
		at org.junit.Assert.assertEquals(Assert.java:647)
		at org.junit.Assert.assertEquals(Assert.java:633)
		at org.opensearch.index.shard.RemoteStoreRefreshListenerTests.lambda$testRefreshSuccessOnSecondAttempt$2(RemoteStoreRefreshListenerTests.java:269)
		at org.opensearch.test.OpenSearchTestCase.assertBusy(OpenSearchTestCase.java:1073)
		... 40 more
```

**Comment by reta** [comment]
> Flaky tests:

I reopened https://github.com/opensearch-project/OpenSearch/issues/8549 :(

**Review by reta — APPROVED** [review]

**Review by Poojita-Raj — APPROVED** [review]

**Review by kotwanikunal — APPROVED** [review]

**Comment by saratvemulapalli** [comment]
This change was not backported to 2.x and it uses a transitive dependency of `com.google.guava:guava:30.1-android` which has an active CVE: https://nvd.nist.gov/vuln/detail/CVE-2023-2976

**Pull request merged**